### PR TITLE
feat(firebase_in_app_messaging)!: federate & rework plugin

### PR DESCRIPTION
**Pull request by Ehesp**

### CODE REVIEW

File: packages/firebase_in_app_messaging/firebase_in_app_messaging_platform_interface/lib/firebase_in_app_messaging_platform_interface.dart [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: require_trailing_commas
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
```
Comment (Salakar) [inline packages/firebase_in_app_messaging/firebase_in_app_messaging_platform_interface/lib/firebase_in_app_messaging_platform_interface.dart:8]:
```suggestion
```

Comment (pr-Mais) [reply packages/firebase_in_app_messaging/firebase_in_app_messaging_platform_interface/lib/firebase_in_app_messaging_platform_interface.dart:8]: what's different?

Comment (Salakar) [reply packages/firebase_in_app_messaging/firebase_in_app_messaging_platform_interface/lib/firebase_in_app_messaging_platform_interface.dart:8]: there's 2 license headers, I'm removing one;

File: packages/firebase_in_app_messaging/firebase_in_app_messaging_platform_interface/lib/src/platform_interface/platform_interface_firebase_in_app_messaging.dart [hunk 2]
```diff
@@ -0,0 +1,70 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: require_trailing_commas
```
Comment (Salakar) [inline packages/firebase_in_app_messaging/firebase_in_app_messaging_platform_interface/lib/src/platform_interface/platform_interface_firebase_in_app_messaging.dart:5]:
```suggestion
```

Comment (Salakar) [reply packages/firebase_in_app_messaging/firebase_in_app_messaging_platform_interface/lib/src/platform_interface/platform_interface_firebase_in_app_messaging.dart:5]: There's a couple other `// ignore_for_file: require_trailing_commas` in some other files too that we should probably remove

**Review by pr-Mais — APPROVED** [review]

**Review by Salakar — APPROVED** [review]

**Pull request merged**